backend/src/main/java/heartbeat/service/report/GenerateReporterService.java

```diff
@@ -129,21 +148,33 @@ private void fetchDataFromKanban(GenerateReportRequest request) {
 				jiraBoardSetting.getBoardColumns(), jiraBoardSetting.getUsers());
 	}
 
-	private void fetchGithubData() {
-		// todo:add fetchGithubData logic
+	private void fetchGithubData(GenerateReportRequest request) {
+		fetchBuildKiteData(request.getStartTime(), request.getEndTime(), request.getCodebaseSetting().getLeadTime(),
+				request.getBuildKiteSetting().getToken());
+		Map<String, String> repoMap = getRepoMap(request.getCodebaseSetting());
+		this.leadTimes = gitHubService
+			.fetchPipelinesLeadTime(this.deployTimesList, repoMap, request.getCodebaseSetting().getToken())
+			.join();
 	}
 
 	private void fetchBuildKiteData(GenerateReportRequest request) {
+		fetchBuildKiteData(request.getStartTime(), request.getEndTime(),
```

Comment:
The two method names are the same, is it not easy to distinguish?  `fetchBuildKiteData `

Reply:
No confusion, their functions are the same, but the parameters are different